cmake/setup_cached_variables.cmake

```diff
@@ -136,6 +136,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
+#
+IF(CMAKE_BUILD_TYPE MATCHES "Release")
```

Comment:
I'm going to add the same comment I had previously: this is hard to read. I'd be much happier if this read
```
  IF(CMAKE_BUILD_TYPE MATCHES "(Debug)?Release")
```
or whatever the appropriate syntax is.